aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -568,6 +573,40 @@ public synchronized void terminate() throws DataStoreException {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void clear(@NonNull Action onComplete,
+                                   @NonNull Consumer<DataStoreException> onError) {
+        try {
+            LOG.debug("Shutting down thread pool for the storage adapter.");
+            threadPool.awaitTermination(THREAD_POOL_TERMINATE_TIMEOUT, TimeUnit.MILLISECONDS);
+            LOG.debug("Storage adapter thread pool shutdown.");
+        } catch (InterruptedException exception) {
+            LOG.warn("Storage adapter thread pool was interrupted during shutdown.", exception);
+        }
+        sqliteStorageHelper.close();
+        databaseConnectionHandle.close();
+        LOG.debug("Clearing DataStore.");
+        if (!context.deleteDatabase(DATABASE_NAME)) {
+            DataStoreException dataStoreException = new DataStoreException(
+                "Error while trying to clear data from the local DataStore storage.",
+                "See attached exception for details.");
+            onError.accept(dataStoreException);
+        }
+        LOG.debug("DataStore cleared. Re-initializing storage adapter.");
+
+        //Re-initialize the adapter.
+        initialize(context, schemaList -> {
+            onComplete.call();
+        }, exception -> {
+                onError.accept(new DataStoreException("Error occurred whilte trying to re-initialize " +
```

Comment:
maybe move the string to a new line instead of concatenating?

Reply:
Done.